tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -50,6 +50,41 @@ static struct k_sem sync_sema;
 static struct k_sem dummy_sema;
 static struct k_thread *main_thread;
 
+/* action for self-defined assert handler. */
+static ZTEST_DMEM volatile bool valid_assert;
+static inline void set_assert_valid(bool valid)
+{
+	valid_assert = valid;
+}
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
+#endif
+{
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+	ARG_UNUSED(file);
+	ARG_UNUSED(line);
+#endif
+	if (valid_assert) {
+		valid_assert = false; /* reset back to normal */
+		printk("Assert error expected as part of test case.\n");
+
+		/* do some action after fatal error happened */
+		ztest_test_pass();
+	} else {
+		printk("Assert failed was unexpected, aborting...\n");
+#ifdef CONFIG_USERSPACE
+	/* User threads aren't allowed to induce kernel panics; generate
+	 * an oops instead.
+	 */
+	if (_is_user_context()) {
+		k_oops();
+	}
+#endif
+		k_panic();
+	}
+}
```

Comment:
Please remove all codes of testing assert().

Reply:
OK, I'm working on some modules one by one, I'll use enjia's handler instead of this.